src/hooks/useCollections.ts

```diff
@@ -56,34 +51,28 @@ export default function useCollections(opts?: {
   );
 
   const getCollections = useCallback(
-    (offset: number, limit: number) =>
-      debounce(() => _getCollections(offset, limit))(),
+    (url?: string) => debounce(() => _getCollections(url))(),
     [_getCollections]
   );
 
-  const nextPage = useCallback(() => {
-    setOffset(offset + limit);
-  }, [offset, limit]);
-
-  const prevPage = useCallback(() => {
-    setOffset(offset - limit);
-  }, [offset, limit]);
+  const nextPage = useCallback(() => setCurrentUrl(nextUrl), [nextUrl]);
+  const prevPage = useCallback(() => setCurrentUrl(prevUrl), [prevUrl]);
 
   useEffect(() => {
-    if (stacApi && !error && !collections) {
-      getCollections(offset, limit);
+    if (stacApi && !error) {
+      getCollections(currentUrl);
     }
-  }, [getCollections, stacApi, collections, error, offset, limit]);
+  }, [getCollections, stacApi, error, currentUrl]);
 
   return {
     collections,
     reload: useCallback(
-      () => getCollections(offset, limit),
-      [getCollections, offset, limit]
+      () => getCollections(currentUrl),
+      [getCollections, currentUrl]
     ),
-    nextPage: hasNext ? nextPage : undefined,
-    prevPage: hasPrev ? prevPage : undefined,
-    setOffset,
+    nextPage: nextUrl ? nextPage : undefined,
+    prevPage: prevUrl ? prevPage : undefined,
+    setCurrentUrl,
```

Comment:
Do we need to expose this? It feels like this should only be used internally.

Reply:
Actually, thinking about it. Do we need the `currentUrl` state at all? It seems like it's only needed the set the prev/next page URLs, but couldn't we call `getCollections` directly from the `nextPage`/`prevPage` callbacks?

```js
const nextPage = useCallback(() => getCollections(nextUrl), [nextUrl]);
const prevPage = useCallback(() => getCollections(prevUrl), [prevUrl]);
```

Reply:
It is also being used for the reload.